scipy/signal/_sigtoolsmodule.c

```diff
@@ -1086,26 +1086,17 @@ static PyObject *_sigtools_convolve2d(PyObject *NPY_UNUSED(dummy), PyObject *arg
             afill = (PyArrayObject *)PyArray_Cast(tmp, typenum);
             Py_DECREF(tmp);
             if (afill == NULL) goto fail;
-            /* Deprecated 2017-07, scipy version 1.0.0 */
-            if (DEPRECATE("could not cast `fillvalue` directly to the output "
-                          "type (it was first converted to complex). "
-                          "This is deprecated and will raise an error in the "
-                          "future.") < 0) {
-                goto fail;
-            }
+            PYERR("could not cast `fillvalue` directly to the output "
+                  "type (it was first converted to complex).");
         }
         if (PyArray_SIZE(afill) != 1) {
             if (PyArray_SIZE(afill) == 0) {
                 PyErr_SetString(PyExc_ValueError,
                                 "`fillvalue` cannot be an empty array.");
                 goto fail;
```

Comment:
This could be cleaned up as bellow but since it is not strictly part of this pr wasn't sure if it was best just to leave alone.
```suggestion
                PYERR("`fillvalue` cannot be an empty array.");
```